src/handling_of_spatial_data.Rmd

```diff
@@ -0,0 +1,123 @@
+
+For performance assessment, a repeated spatial cross-validation with 5 folds and 10 repetitions is chosen.
+
+### Spatial Cross-Validation
+
+```{r, eval = TRUE}
+data("bc.task.spatial")
+bc.task.spatial
+
+learner.glm = makeLearner("classif.ranger", predict.type = "prob")
+
+resampling = makeResampleDesc("SpRepCV", fold = 5, reps = 10)
+
+set.seed(123)
+out = resample(learner = learner.glm, task = bc.task.spatial,
+  resampling = resampling, measures = list(auc))
+
+mean(out$measures.test$auc)
+```
+
+We can check for the introduced spatial autocorrelation bias here by performing the same modeling task using a non-spatial partitioning setting.
+To force non-spatial cross-validation here, variables `x` and `y` need to be removed from the data set because they are not used for partitioning when using `RepCV` and also should not be used as predictors. 
+
+Additionally, the `spatial` task attribute needs to be set to `FALSE`.
+`mlr` will issue a warning message if it detects variables named `x` or `y` in combination with  `spatial = FALSE`.
+Finally, `RepCV` is chosen instead of `SpRepCV`.
+
+### Non-Spatial Cross-Validation
+
+```{r, eval = TRUE}
+data("bc.task.spatial")
+bc.task.spatial
+
+bc.task.spatial$task.desc$spatial = FALSE
+bc.task.spatial$env$data$x = NULL
+bc.task.spatial$env$data$y = NULL
+
+learner.glm = makeLearner("classif.ranger", predict.type = "prob")
+
+resampling = makeResampleDesc("RepCV", fold = 5, reps = 10)
+
+set.seed(123)
+out = resample(learner = learner.glm, task = bc.task.spatial,
+  resampling = resampling, measures = list(auc))
+
+mean(out$measures.test$auc)
+```
+
+The introduced bias (caused by spatial autocorrelation) in performance in this example is around 0.07 AUROC. 
+
+## Notes
+
+* Usually, coordinates are not used as predictors in spatial modeling.
+If you insist on using them and also want to perform spatial cross-validation, add them to the data set with a different name than `x` or `y`, e.g. `x1` and `y1`.
+That way they will be treated as normal predictors.
+
+* Some models are more affected by spatial autocorrelation than others. 
+In general, it can be said that the more flexible a model is, the more it will profit from underlying spatial autocorrelation.
+Simpler models (e.g., GLM) will show less overoptimistic performance estimates.
+
+* The concept of spatial cross-validation was originally implemented in package [sperrorest](https://pat-s.github.io/sperrorest/index.html).
+This package comes with even more partitioning options and the ability to visualize the spatial grouping of folds. 
+We plan to integrate more functions from [sperrorest](https://pat-s.github.io/sperrorest/index.html) into `mlr` so stay tuned!
+
+* For more detailed information, see [Brenning 2005](https://www.nat-hazards-earth-syst-sci.net/5/853/2005/), [Brenning2012](http://ieeexplore.ieee.org/document/6352393/), Schratz2018 (in preperation).
```

Comment:
Please don't reference something that can't be accessed -- either remove Schratz2018 and add it when available, or provide a link to a preliminary copy.